Sources/Core/AWSClientRuntime/UserAgent/AWSUserAgentMetadata.swift

```diff
@@ -73,10 +73,10 @@ public struct AWSUserAgentMetadata {
         self.frameworkMetadata = frameworkMetadata
     }
 
-    public static func fromConfig<ServiceSpecificConfiguration>(
+    public static func fromConfig(
         serviceID: String,
         version: String,
-        config: AWSClientConfiguration<ServiceSpecificConfiguration>
+        config: DefaultClientConfiguration & AWSDefaultClientConfiguration
```

Comment:
This narrows down service specific configurations to default configurations (which I think is good).

But is there a use case where we might need service specific configuration?

Reply:
I refactored the `fromConfig` function because 2) `AWSClientConfiguration ` and 2) the function only uses variables specified by the two configuration protocols. (it doesn't use any service specific stuff)